src/ai/backend/manager/api/gql/model_deployment/model_deployment.py

```diff
@@ -0,0 +1,790 @@
+@strawberry.type
+class DeploymentStatusChangedPayload:
+    deployment: ModelDeployment
+
+
+@strawberry.type
+class ReplicaStatusChangedPayload:
+    replica: ModelReplica
+
+
+@strawberry.type
+class DeploymentMetricsUpdatedPayload:
+    metrics: DeploymentMetrics
+
+
+# Input Types
+@strawberry.input
+class RollingConfigInput:
+    max_surge: int
+    max_unavailable: int
+
+
+@strawberry.input
+class BlueGreenConfigInput:
+    auto_promotion_enabled: bool
+    termination_wait_time: int
+
+
+@strawberry.input
+class CanaryConfigInput:
+    canary_percentage: int
+    canary_duration: str
+    success_threshold: float
+
+
+@strawberry.input
+class DeploymentStrategyInput:
+    type: DeploymentStrategyType
+
+
+@strawberry.input
+class CreateModelDeploymentInput:
+    name: str
+    preferred_domain_name: Optional[str] = None
+    open_to_public: bool
+    tags: Optional[list[str]] = None
+    cluster_config: ClusterConfigInput
+    deployment_strategy: DeploymentStrategyInput
+    initial_revision: CreateModelRevisionInput
+
+
+@strawberry.input
+class UpdateModelDeploymentInput:
+    id: ID
+    open_to_public: Optional[bool] = None
+    tags: Optional[list[str]] = None
+    deployment_strategy: Optional[DeploymentStrategyInput] = None
+    active_revision_id: Optional[ID] = None
+
+
+@strawberry.input
+class DeleteModelDeploymentInput:
+    id: ID
+    force_delete: bool = False
+
+
+# Connection types for Relay support
+@strawberry.type
+class ModelDeploymentConnection(Connection[ModelDeployment]):
+    @strawberry.field
+    def count(self) -> int:
+        return 0
+
+    @classmethod
+    def resolve_connection(
+        cls,
+        nodes: NodeIterableType[ModelDeployment],
+        *,
+        info: Info,
+        before: Optional[str] = None,
+        after: Optional[str] = None,
+        first: Optional[int] = None,
+        last: Optional[int] = None,
+        max_results: Optional[int] = None,
+        **kwargs,
+    ) -> "ModelDeploymentConnection":
+        return cls(
+            edges=[],
+            page_info=relay.PageInfo(
+                has_next_page=False,
+                has_previous_page=False,
+                start_cursor=None,
+                end_cursor=None,
+            ),
+        )
+
+
+# Resolvers
+@strawberry.relay.connection(ModelDeploymentConnection)
+async def deployments(
+    filter: Optional[DeploymentFilter] = None,
+    order_by: Optional[DeploymentOrderBy] = None,
+    first: Optional[int] = None,
+    after: Optional[str] = None,
+) -> list[ModelDeployment]:
+    """List deployments with optional filtering and pagination."""
+    # Return a list of mock deployments with more details
+    model_names = [
+        "llama-3-8b-instruct",
+        "mistral-7b-v0.3",
+        "gemma-2-9b",
+        "qwen-2.5-7b-instruct",
+        "mixtral-8x7b",
+    ]
+
+    deployments = []
+    for i, name in enumerate(model_names):
+        deployment = ModelDeployment(
+            id=ID(f"dep-{i + 1:03d}"),
```

Comment:
Is there any rule of ID type value?

Reply:
We must use following id generation logic. (`{domain}:{id}`) @seedspirit